mvc/src/main/java/com/interface21/webmvc/servlet/mvc/handler/ModelAttributeArgumentResolver.java

```diff
@@ -0,0 +1,74 @@
+package com.interface21.webmvc.servlet.mvc.handler;
+
+import com.interface21.web.bind.annotation.ModelAttribute;
+import com.interface21.web.bind.annotation.RequestMethod;
+import com.interface21.web.http.MediaType;
+import com.interface21.webmvc.servlet.mvc.support.HttpRequestBodyParser;
+import com.interface21.webmvc.servlet.mvc.support.QueryStringParser;
+import com.interface21.webmvc.servlet.mvc.support.TypeConversionUtil;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.lang.reflect.*;
+import java.util.Arrays;
+import java.util.Map;
+
+public class ModelAttributeArgumentResolver implements ArgumentResolver {
+    @Override
+    public boolean supports(Parameter parameter) {
+        return parameter.isAnnotationPresent(ModelAttribute.class);
+    }
+
+    @Override
+    public Object resolve(Parameter parameter, Method method, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        String queryString = getQueryString(request);
+        Map<String, String> parsedQueryString = QueryStringParser.parse(queryString);
+
+        Class<?> parameterType = parameter.getType();
+        Object parameterTypeInstance = createParameterTypeInstance(parameterType);
+        setUpInstanceFieldValue(parsedQueryString, parameterType, parameterTypeInstance);
+        return parameterTypeInstance;
+    }
+
+    private String getQueryString(HttpServletRequest request) throws IOException {
+        if (request.getMethod().equals(RequestMethod.POST.name()) && request.getHeader("Content-Type").equals(MediaType.FORM_URL_ENCODED)) {
```

Comment:
매개변수의 내부값을 주체로 비교할 경우 NPE 이슈에서 자유로울 수 없습니다. 반대로 비교하거나 아니면
다음과 같이 해보는것도 괜찮을 것 같네요.
```suggestion
        RequestMethod targetMethod = RequestMethod.from(request.getMethod());
        MediaType targetMediaType = MediaType.from(request.getHeader("Content-Type"));
        if RequestMethod.POST == targetMediaType && MediaType.FORM_URL_ENCODED == targetMediaType) {
```